system/Entity/Cast/BooleanCast.php

```diff
@@ -23,4 +23,12 @@ public static function get($value, array $params = []): bool
     {
         return (bool) $value;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function set($value, array $params = []): bool
+    {
+        return filter_var($value, FILTER_VALIDATE_BOOLEAN);
```

Comment:
Can you add FILTER_NULL_ON_FAILURE here as well. I've made similar PR while ago but it was not merged in order to keep casting work similarly as other frameworks do it.

https://github.com/codeigniter4/CodeIgniter4/pull/4591

Reply:
@najdanovicivan Can you show how other frameworks do it?